lib/ansible/modules/cloud/openstack/os_keystone_endpoint.py

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/python
+# Copyright (c) 2017 VEXXHOST, Inc.
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: os_keystone_endpoint
+short_description: Manage OpenStack Identity service endpoints.
+extends_documentation_fragment: openstack
+author: "Mohammed Naser (@mnaser)"
+version_added: "2.4"
+description:
+    - Create, update, or delete OpenStack Identity service endpoints. If a
+      service with the same combination of I(service), I(interface) and I(region)
+      exist, the I(url) and I(state) (C(present) or C(absent)) will be updated.
+options:
+   service:
+     description:
+        - Name of the service.
+     required: true
+   interface:
+     description:
+        - Interface of the service.
+     choices: [admin, public, internal]
+     required: true
+   url:
+     description:
+        - URL of the service.
+     required: true
+   region:
+     description:
+        - Region that the service belongs to. Note that I(region_name) is used for authentication.
+     required: true
+   enabled:
+     description:
+        - Is the service enabled.
+     default: True
+   state:
+     description:
+       - Should the resource be C(present) or C(absent).
+     choices: [present, absent]
+     default: present
+requirements:
+    - "python >= 2.6"
+    - "shade"
```

Comment:
If requirement versions are specified here does that get used by a build system anywhere or is this purely a docstring for the user?

Reply:
Good question! I think we could remove `python >= 2.6` since the developing new modules page states:
```
Adding Python 3 support to modules (all new modules must be Python-2.6 and Python-3.5 compatible).
```

Reply:
I guess this is just docstring for the user